src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1430,8 +1430,10 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
         continue;
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
-      if (!failureMessage.isPresent()) {
```

Comment:
No my concern was that you presented a solution to an outdated problem which is a waste of time. 

- Not sure how you are measuring the method length but looks about the same to me
- Not even sure what point you are making here as just like any code you read parts of it and build up to the whole just like the stream/filter
- I would bet it takes most developers far longer to understand the filter mess you have above than the original code

I'm fine with presenting refactoring ideas but I disagree in this case and think we should move forward.